File: VB/WindowsFormsApplication741/MyFlyoutPanelToolFormHandler.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.Utils
Imports DevExpress.Utils.Win

Namespace WindowsFormsApplication741
    Public Class MyFlyoutPanelToolFormHandler
        Inherits FlyoutPanelToolFormHandler

        Public Sub New(ByVal toolForm As FlyoutPanelToolForm)
            MyBase.New(toolForm)
        End Sub
        Protected Overrides Function CreateAnimationProvider() As PopupToolWindowAnimationProviderBase
            If Anchor = PopupToolWindowAnchor.Left Then
                Return New MyPopupToolWindowLeftRightSlideAnimationProvider(Me)
            End If
            Return MyBase.CreateAnimationProvider()
        End Function
        Protected ReadOnly Property Anchor() As PopupToolWindowAnchor
            Get
                Return DirectCast(Me, IPopupToolWindowAnimationSupports).AnchorType
            End Get
        End Property
    End Class
End Namespace

File: CS/WindowsFormsApplication741/MyFlyoutPanelToolFormHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Utils;
using DevExpress.Utils.Win;

namespace WindowsFormsApplication741 {
    public class MyFlyoutPanelToolFormHandler : FlyoutPanelToolFormHandler {
        public MyFlyoutPanelToolFormHandler(FlyoutPanelToolForm toolForm)
            : base(toolForm) {
        }
        protected override PopupToolWindowAnimationProviderBase CreateAnimationProvider() {
            if(Anchor == PopupToolWindowAnchor.Left) {
                return new MyPopupToolWindowLeftRightSlideAnimationProvider(this);
            }
            return base.CreateAnimationProvider();
        }
        protected PopupToolWindowAnchor Anchor {
            get {
                return ((IPopupToolWindowAnimationSupports)this).AnchorType;
            }
        }
    }
}
